src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.CSharp.Tests.cs

```diff
@@ -0,0 +1,488 @@
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 13, 8, 38, StringComparisonRule, StartsWithName);
+        }
+
+        [Fact]
+        public Task Diagnostic_StartsWith_ToLowerInvariant_If()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.ToLowerInvariant().StartsWith(b))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.StartsWith(b, StringComparison.InvariantCultureIgnoreCase))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 13, 8, 47, StringComparisonRule, StartsWithName);
+        }
+
+        [Fact]
+        public Task Diagnostic_StartsWith_ToUpper_If()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.ToUpper().StartsWith(b))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.StartsWith(b, StringComparison.CurrentCultureIgnoreCase))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 13, 8, 38, StringComparisonRule, StartsWithName);
+        }
+
+        [Fact]
+        public Task Diagnostic_StartsWith_ToUpperInvariant_If()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.ToUpperInvariant().StartsWith(b))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.StartsWith(b, StringComparison.InvariantCultureIgnoreCase))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 13, 8, 47, StringComparisonRule, StartsWithName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToLower_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToLower().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.CurrentCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
```

Comment:
I think fixed code changes semantics.

```csharp
using System;

string a = "a";
string b = "A";

Console.WriteLine(a.ToLower().CompareTo(b)); // Compares "a" to "A", gives -1
Console.WriteLine(StringComparer.CurrentCultureIgnoreCase.Compare(a, b)); // Ignores case, so gives 0
```

Reply:
I asked the question in the main issue https://github.com/dotnet/runtime/issues/78606#issuecomment-1572625053

Reply:
Ok, so for now, I'll only diagnose `CompareTo`, but won't offer a fix. We can address this later if we want to.

Reply:
@Youssef1313 what's the best way to determine if an analyzer triggered, but nothing got fixed? Like this?:

```cs
        private Task YesDiagnosticButNoFix(string originalSource)
        {
            VerifyVB.Test test = new()
            {
                TestCode = originalSource,
                FixedCode = originalsource
            };
            return test.RunAsync();
        }
```

The reason I ask is because I am looking at other examples, and it seems that when we don't expect a _diagnostic_ at all, we just pass a `TestCode` string, but no `FixedCode`, like this:

```cs
        private Task NoDiagnostic(string originalSource)
        {
            VerifyVB.Test test = new()
            {
                TestCode = originalSource,
            };
            return test.RunAsync();
        }
```

Reply:
When there is no diagnostic, both snippets should be equivalent.
When there is a diagnostic, the first should be used as it asserts that no code fix is offered

Reply:
Ah okay, my examples were inverted. Thanks for the confirmation.